kaska/s2_observations.py

```diff
@@ -196,39 +238,96 @@ def read_time_series(self, time_grid):
         return s2_obs
 
     def read_granule(self, timestep):
-        """NOTE: Currently reads in sequentially. It's better to gather
-        all the filenames and read them in parallel using parmap.py"""
-        current_folder = self.date_data[timestep]
-
-        fname_prefix = [
-            f.name.split("B02")[0] for f in current_folder.glob("*B02_sur.tif")
-        ][0]
-        cloud_mask = current_folder.parent / f"cloud.tif"
-        
-        cloud_mask = reproject_data(
-            str(cloud_mask), target_img=self.state_mask
-        ).ReadAsArray()
-        mask = cloud_mask <= self.band_prob_threshold
-        if mask.sum() == 0:
-            # No pixels! Pointless to carry on reading!
-            LOG.info("No clear observations")
-            return None, None, None, None, None, None
+        """Reads the data for the given timestamp.
 
+        Parameters
+        ----------
+        timestep : A date.
+        Returns
+        -------
+        tuple containing the rho_surface, mask, sza, vza, raa, rho_unc
+
+        NOTE: Currently reads in sequentially. It's better to gather
+        all the filenames and read them in parallel using parmap.py
+        """
+        current_folder = self.date_data[timestep][0].parent.parent
+
+        # Read in cloud mask.
+        cloud_mask = current_folder / f"cloud.tif"
+        # Fish out state_mask shape and make dummy (i.e. all True)
+        # cloud mask of the same shape.
+        shape_l = [i.split(" ")
+                   for i in gdal.Info(self.state_mask).split("\n")
+                   if "Size is " in i][0]
+        shape_l = [int(i.split(",")[0]) for i in shape_l[2:]]
+        shape_l.reverse()
+        shape = tuple(shape_l)
+        mask = np.ones(shape, dtype=np.bool)
+        # Reproject cloud mask on state mask.
+        # Stop processing if no clear pixels exist.
+        if cloud_mask.exists():
+            cloud_mask = reproject_data(
+                str(cloud_mask), target_img=self.state_mask
+            ).ReadAsArray()
+            mask = cloud_mask <= self.band_prob_threshold
+            if mask.sum() == 0:
+                # No pixels! Pointless to carry on reading!
+                LOG.info("No clear observations")
+                return None, None, None, None, None, None
+        else:
+            LOG.info(f"Cloud file {cloud_mask} does not exist. No cloud mask" +
```

Comment:
Hm, thanks! I think I like the first option more, cause you can freely move stuff between lines, without worrying about all the `""`'s and `f`'s...

Reply:
but then the indentation is weird, because they won't be on the same line, but 
with blank spaces at the begininning if you want to leave the code "readable".